### ISSUE TITLE
Delete file call failed

### ISSUE THREAD

**Issue by yaoqiangpersonal** [Raudius/files_scripts#161]
When I delete a matching non-root file, I get the error "Could not find input file belonging in root folder for file action"

**Comment by yaoqiangpersonal** [comment]
I run it from flow, as follows
![屏幕截图 2024-03-21 022938](https://github.com/Raudius/files_scripts/assets/32598124/b6c78430-a503-4762-ac5a-7c15428b7950)


script
```lua
function trigger()
local a = http_request("http://xxxx",'GET')
  local user = users_find(nil,"xx")[1]
    if a ~= nil then
      notify(user,"sync",a)
    end
end

trigger()
```
When I add an xx.mp3 file to the /music folder, everything works fine, but when I delete the file an error occurs. I checked nextcloud’s logs and found “Could not find input file belonging in root folder for file action”

**Comment by yaoqiangpersonal** [comment]
```json
{
"reqId":"JCDu3tMjR6gz1SxHnyzC",
"level":3,"time":"2024-03-20T18:25:29+00:00","remoteAddr":"172.18.0.4",
"user":"weiliang",
"app":"files_scripts",
"method":"DELETE",
"url":"/remote.php/dav/files/weiliang/music/Finale.mp3",
"message":"Could not find input file belonging in root folder for file action",
"userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36",
"version":"28.0.3.2",
"data":{
  "app":"files_scripts",
  "root":"/weiliang/files",
  "path":"/music/Finale.mp3"
},
"id":"65fbacd714949"}
```

Here are the details of the log

**Comment by Raudius** [comment]
This happens because when the flow triggers the file is already deleted. It tries to fetch the file but fails. It needs it to create the interpeter context which would allow you to view the related file via `get_input_files()`

What would be your preferred way to handle this? I assume you would still like to have the path to the deleted file in your action :thinking: 

It might be possible to create a fake file object that we can still return via `get_input_files()` or we can include the filepath separately as a variable `flow_deleted_file`

**Comment by yaoqiangpersonal** [comment]
> This happens because when the flow triggers the file is already deleted. It tries to fetch the file but fails. It needs it to create the interpeter context which would allow you to view the related file via `get_input_files()`
> 
> What would be your preferred way to handle this? I assume you would still like to have the path to the deleted file in your action 🤔
> 
> It might be possible to create a fake file object that we can still return via `get_input_files()` or we can include the filepath separately as a variable `flow_deleted_file`

I think `flow_deleted_file` may be a better way, `get_input_files` will give people an illusion, and `flow_deleted_file` is the explanation of this illusion.

**Comment by Raudius** [comment]
This will be fixed by #167 

I am not planning to backport this fix to older versions of the app, so the fix will only be available in NC 28+
